lib/config.js

```diff
@@ -84,11 +78,6 @@ export default class Config {
     publicSigningKey = '',
     privateSigningKey = ''
   ) {
-    if (!clientId || !apiKeyId || !apiSecret || !publicKey || !privateKey) {
```

Comment:
Is getting rid of both this and (the incorrect) line 59 going to leave us without any client-side validation?

Reply:
No -- the concept of config is becoming a little more tricky. The idea is validation moves a little down the stack. Rather than directly when creating the config object, if moves to where that config is used, allowing config to be used in different contexts and validated for each of them. This philosophy change is documented in the internal docs for SDK approach.

The validation currently is happening at request time - not so much at instantiation time. That should change, but I didn't include that as part of this refactor. The reason this is pulled out here -- a couple changes made are requiring alternate config set ups that would not have been allowed by this validation. More to come, but trying to somewhat break up change-sets.